src/Symfony/Component/Intl/NumberFormatter/NumberFormatter.php

```diff
@@ -296,23 +296,22 @@ public function __construct(?string $locale = 'en', int $style = null, $pattern
      * @throws MethodArgumentValueNotImplementedException When the $style is not supported
      * @throws MethodArgumentNotImplementedException      When the pattern value is different than null
      */
-    public static function create($locale = 'en', $style = null, $pattern = null)
+    public static function create(?string $locale = 'en', int $style = null, string $pattern = null)
     {
         return new self($locale, $style, $pattern);
     }
 
     /**
      * Format a currency value.
      *
-     * @param float  $value    The numeric currency value
      * @param string $currency The 3-letter ISO 4217 currency code indicating the currency to use
      *
      * @return string The formatted currency value
      *
      * @see http://www.php.net/manual/en/numberformatter.formatcurrency.php
      * @see https://en.wikipedia.org/wiki/ISO_4217#Active_codes
      */
-    public function formatCurrency($value, $currency)
+    public function formatCurrency(float $value, string $currency)
```

Comment:
I‘m not sure if we should actually force `float` here. Storing a decimal value as string is a valid pattern, especially when it comes to currency amounts.

Reply:
AFAIU, this `NumberFormatter` implementation is a sort of polyfill for `Intl::NumberFormatter` where `$value`  is typed as float and generates the following error if passed as a string:
`Fatal error: Uncaught TypeError: NumberFormatter::formatCurrency() expects parameter 1 to be float, string given in`

So question is, should we accept a wider range of type compared to the implementation it's trying to polyfill ? 

Personally I would say no, I would expect this function to also throws a TypeError on string arg if the whole point of this class is to be as close as the "real" one.

I'll leave it type-hinted as a float for now, open to modification if team says otherwise.

Reply:
All right, thanks for checking.